pkg/probe/util_test.go

```diff
@@ -41,3 +42,103 @@ func TestFindPortByName(t *testing.T) {
 		t.Errorf("Expected %v, got %v, err: %v", want, got, err)
 	}
 }
+
+func TestResolveContainerPort(t *testing.T) {
+	type args struct {
+		param     intstr.IntOrString
+		container *v1.Container
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    int
+		wantErr bool
+	}{
+		{
+			name: "get port by int type",
+			args: args{
+				param:     intstr.IntOrString{Type: 0, IntVal: 443},
+				container: &v1.Container{},
+			},
+			want:    443,
+			wantErr: false,
+		},
+		{
+			name: "invalid port",
+			args: args{
+				param:     intstr.IntOrString{Type: 0, IntVal: 66666},
+				container: &v1.Container{},
+			},
+			want:    66666,
+			wantErr: true,
+		},
+		{
+			name: "get port by port name",
+			args: args{
+				param: intstr.IntOrString{Type: 1, StrVal: "foo"},
+				container: &v1.Container{
+					Ports: []v1.ContainerPort{
+						{
+							Name:          "foo",
+							ContainerPort: 8080,
+						},
+						{
+							Name:          "bar",
+							ContainerPort: 9000,
+						},
+					},
+				},
+			},
+			want:    8080,
+			wantErr: false,
+		},
+		{
+			name: "no port name",
+			args: args{
+				param: intstr.IntOrString{Type: 1, StrVal: "foo"},
+				container: &v1.Container{
+					Ports: []v1.ContainerPort{
+						{
+							Name:          "bar",
+							ContainerPort: 9000,
+						},
+					},
+				},
+			},
+			want:    0,
+			wantErr: true,
+		},
+		{
+			name: "invalid param type",
+			args: args{
+				param: intstr.IntOrString{Type: 2, StrVal: "foo"},
+				container: &v1.Container{
+					Ports: []v1.ContainerPort{
+						{
+							Name:          "foo",
+							ContainerPort: 8080,
+						},
+						{
+							Name:          "bar",
+							ContainerPort: 9000,
+						},
+					},
+				},
+			},
+			want:    -1,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := ResolveContainerPort(tt.args.param, tt.args.container)
```

Comment:
Can we run those in parallel?
```suggestion
			t.Parallel()
			got, err := ResolveContainerPort(tt.args.param, tt.args.container)
```